pkg/attribute/protoBag.go

```diff
@@ -33,14 +33,28 @@ type ProtoBag struct {
 	messageDict         map[string]int32
 	convertedStringMaps map[int32]map[string]string
 	stringMapMutex      sync.RWMutex
+
+	// to keep track of attributes that are referenced
+	referencedAttrs      map[string]mixerpb.ReferencedAttributes_Condition
```

Comment:
Eventually. For now, syncmap doesn't support a Len method which is really a bummer.